src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private ArrayList<String> readFile(String fileName) {
+        File fileToRead = new File(fileName);
+
+        try (BufferedReader br = new BufferedReader(new FileReader(fileToRead))) {
+            ArrayList<String> operationsList = new ArrayList<>();
+            String line;
+
+            while ((line = br.readLine()) != null) {
+                operationsList.add(line);
+            }
+
+            return operationsList;
+        } catch (IOException e) {
+            throw new RuntimeException("File not found: ", e);
+        }
+    }
+
+    private String[] createReport(ArrayList<String> operationsList) {
+        int supplyOperations = 0;
+        int buyOperations = 0;
+
+        for (String operation : operationsList) {
+            String[] partsOfOperation = operation.split(",");
+            String type = partsOfOperation[0];
+            int quantity = Integer.parseInt(partsOfOperation[1]);
 
+            if (type.equals("buy")) {
+                buyOperations += quantity;
+            } else {
+                supplyOperations += quantity;
+            }
+        }
+
+        return new String[]{
+                "supply," + supplyOperations,
+                "buy," + buyOperations,
+                "result," + (supplyOperations - buyOperations)
+        };
+    }
+
+    private void writeDataToFile(String fileName, String[] operations) {
```

Comment:
The method 'writeDataToFile' should be named to reflect its purpose more clearly, such as 'writeLinesToFile'.